pkg/cmd/cli/cmd/env.go

```diff
@@ -268,14 +285,14 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 		}
 	}
 	if one && skipped == len(infos) {
-		return fmt.Errorf("the %s %s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
+		return fmt.Errorf("%s/%s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
 	}
 
 	if list {
 		return nil
 	}
 
-	objects, err := resource.AsVersionedObject(infos, false, outputVersion)
+	objects, err := resource.AsVersionedObjects(infos, outputVersion)
```

Comment:
Previously, we could do `--output-version=v1beta3`, get v1beta3 objects, then successfully pass those to the API (since `resource.NewHelper(info.Client, info.Mapping).Replace(...)` would convert the v1beta3 object to a v1 object if that was what the client had negotiated with the server).

Now, however, we're calling `resource.NewHelper(info.Client, info.Mapping).Patch(...)` and passing []byte content, which means the burden is on us to compute the patch using the same API version that the client will use to talk to the server.

I think that means the input the the patch computation needs to be objects versioned to the result of `--api-version -> clientConfig.Version -> server negotiation`